torch/custom_class.h

```diff
@@ -163,40 +171,22 @@ class class_ {
  private:
   template <typename Func>
   void defineMethod(std::string name, Func func) {
-    auto graph = std::make_shared<Graph>();
-    auto qualFuncName = className + "::" + name;
-    ensure_c10_registerer_defined();
-    registeredOps().push_back(
-        torch::RegisterOperators().op(qualFuncName, std::move(func)));
-    auto func_symbol = c10::Symbol::fromQualString(qualFuncName);
-    auto ops = torch::jit::getAllOperatorsFor(func_symbol);
-    TORCH_CHECK(ops.size() == 1);
-    auto &schema = ops[0]->schema();
-
-    for (const auto& arg : schema.arguments()) {
-      graph->addInput()->setType(arg.type());
-    }
-
-    auto opCall = graph->insertNode(graph->create(
-        func_symbol, graph->inputs(), schema.returns().size()));
-    Value* res;
-    if (schema.returns().size() > 1) {
-      const auto& returns = schema.returns();
-      size_t op_invocation_idx = 0;
-      for (const auto& ret : returns) {
-        opCall->output(op_invocation_idx++)->setType(ret.type());
-      }
-      res = graph->insertNode(graph->createTuple(opCall->outputs()))->output();
-    } else if (schema.returns().size() == 1) {
-      const auto& returns = schema.returns();
-      res = opCall->output()->setType(returns[0].type());
-    } else {
-      res = graph->insertConstant(IValue())->setType(NoneType::get());
-    }
-    graph->registerOutput(res);
-
-    auto method = classCU()->create_function(qualClassName + "." + name, graph);
-    classTypePtr->addMethod(method);
+    auto qualMethodName = qualClassName + "." + name;
+    auto schema = c10::inferFunctionSchema<Func>(std::move(name), "");
+    auto wrapped_func = [func = std::move(func)](Stack& stack) mutable -> void {
+      using RetType =
+          typename c10::guts::infer_function_traits_t<Func>::return_type;
+      detail::BoxedProxy<RetType, Func>()(stack, func);
```

Comment:
so the question is how to handle profiling for these ops. If we don't go through dispatcher - maybe we should call it here directly (with something like https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/register_c10_ops.cpp#L18 )

Reply:
I'll just add a RECORD_FUNCTION

Reply:
hot take: we should RECORD_FUNCTION in CALL instructions in the interpreter. That would catch this case and also show what time we are spending in actual methods.